ui/tests/acceptance/saml-auth-method-test.js

```diff
@@ -11,25 +11,25 @@ import { Response } from 'miragejs';
 import { windowStub } from 'vault/tests/helpers/oidc-window-stub';
 import { setupTotpMfaResponse } from 'vault/tests/helpers/mfa/mfa-helpers';
 import { AUTH_FORM } from 'vault/tests/helpers/auth/auth-form-selectors';
+import { logout } from 'vault/tests/helpers/auth/auth-helpers';
 
 module('Acceptance | enterprise saml auth method', function (hooks) {
   setupApplicationTest(hooks);
   setupMirage(hooks);
 
-  hooks.beforeEach(function () {
+  hooks.beforeEach(async function () {
     this.openStub = windowStub();
     this.server.put('/auth/saml/sso_service_url', () => ({
       data: {
-        sso_service_url: 'http://sso-url.hashicorp.com/service',
+        sso_service_url: 'test/fake/sso/route', // we aren't actually opening a popup so use a fake url
         token_poll_id: '1234',
       },
     }));
     this.server.put('/auth/saml/token', () => ({
       auth: { client_token: 'root' },
     }));
     // ensure clean state
-    localStorage.removeItem('selectedAuth');
-    visit('/vault/logout');
+    await logout(); // clears local storage
```

Comment:
Does this make more sense to have in the `afterEach` block?

Reply:
Good question - it's also there, but because we're stubbing the `sys/internal/ui/mounts` request here (which fetches visible mounts in the auth route and displays them as tabs in the UI login form) we want to refresh the login form state.

Reply:
Oh, I answered why I added it to the test...

It's here because the test than ran before this could be a separate module that logged in with a different method so we want to make sure that the form state is fresh for each new saml login attempt.

Reply:
The whole UI test server state is 🥴 since the same vault cluster runs throughout. It's really hard to know if the before or after hook is best for these things. Historically I have noticed that the `afterEach` hook is not as reliable for "cleanup" so it seems to work best to clean up the necessary items before running tests.